### PR TITLE
Added new functions to apoc.coll - shuffle(), randomItem(), randomItems()

### PR DESCRIPTION
**Pull request by InverseFalcon**
This fulfills feature request #277.

### CODE REVIEW

File: src/test/java/apoc/coll/CollTest.java [hunk 1]
```diff
@@ -204,4 +205,138 @@ public void testSetOperations() throws Exception {
         testCall(db,"CALL apoc.coll.removeAll([1,2],[3,2])", r -> assertEquals(asList(1L),r.get("value")));
 
     }
+
```
Comment (InverseFalcon) [inline src/test/java/apoc/coll/CollTest.java]: Any suggestions and improvements for better testing indeterminate behavior are more than welcome.

**Comment by InverseFalcon** [comment]
Any suggestions or improvements for better testing indeterminate behavior are more than welcome.

File: src/main/java/apoc/coll/Coll.java [hunk 2]
```diff
@@ -260,5 +261,49 @@ public static int compare(Object o1, Object o2) {
...
+			return Stream.of(new ObjectResult(null));
+		}
+	
+		Object value = coll.get(ThreadLocalRandom.current().nextInt(coll.size()));
+        return Stream.of(new ObjectResult(value));
+    }
+    
+    @Procedure
+    @Description("apoc.coll.randomItems(coll, itemCount, allowRepick) - returns a list of itemCount random items from the original list, optionally allowing picked elements to be picked again")
+    public Stream<ListResult> randomItems(@Name("coll") List<Object> coll, @Name("itemCount") long itemCount, @Name("allowRepick") boolean allowRepick) {
```
Comment (jexp) [inline src/main/java/apoc/coll/Coll.java]: can we make allowRepick to default to false?

**Comment by jexp** [comment]
Sorry for the delay.
Looks good to me except for my one comment.

**Comment by jexp** [comment]
Oh sorry, I forgot, can you add them also to docs/overview.adoc in the appropriate place?

**Comment by InverseFalcon** [comment]
I'll move these over to the 3.1 branch to use defaults, and update the docs.

**Comment by InverseFalcon** [comment]
Retiring this pull request, will submit a new one on the 3.1 branch.

**Pull request closed without merging**